homeassistant/components/esphome/__init__.py

```diff
@@ -705,6 +705,8 @@ def __init__(
         self._component_key = component_key
         self._key = key
         self._state_type = state_type
+        if entry_data.device_info is not None and entry_data.device_info.friendly_name:
+            self._attr_has_entity_name = entry_data.device_info.friendly_name
```

Comment:
`friendly_name` is typed to be a `str`.  I'm not sure where the mypy error is coming from.

Reply:
Apparently I need more coffee 🙃